### ISSUE TITLE
Production build hangs indefinitely

### ISSUE THREAD

**Issue by zeppelin** [ebryn/ember-ckeditor#10]
```
$ ember -v
version: 0.2.7
node: 1.8.1
npm: 2.11.0
```

Repro steps:

```
$ ember new whatever
$ cd whatever
$ ember install ember-ckeditor
$ ember build -e production
```

The last command hangs with 100% CPU consumption.

**Comment by dustyjewett** [comment]
It will eventually finish (I've seen it take 20-30 mins), but this seems to be caused by the uglify stage + the plugins js files (I suspect the language files). We use a whitelist for including files which isn't a very good option for a generic addon.

```
ckAssets = new Funnel('bower_components/ckeditor', {
    srcDir: '/',
    include: [
        'ckeditor.js',
        'adapters/jquery.js',
        'styles.js',
        'config.js',
        'contents.css',
        'lang/en.js',
        'plugins/widget/plugin.js',
        'plugins/widget/lang/en.js',
        'plugins/autogrow/plugin.js',
        'plugins/**/*.png',
        'plugins/**/*.gif'],
    destDir: '/assets/ckeditor'
});
```

**Comment by jwhitmarsh** [comment]
@dustyjewett sorry to be a pain, what is your complete setup for doing it your way? I've done this but now I just get 'CKEDITOR is not defined', so clearly I have to attach it to the window somewhere, but not sure where.

**Comment by dustyjewett** [comment]
That's the base (we do more because we have our own plugins and skin).  

Make sure you include the tree at the bottom of the brocfile:

```
module.exports = app.toTree([ckAssets]);
```

**Comment by ChrisHonniball** [comment]
I was bitten by this one today as well. I've used @dustyjewett's solution with slight modifications. Still takes a while though.

**Comment by ChrisHonniball** [comment]
Found it to be fingerprinting... Updated `ember-cli-build.js` to include:

``` javascript
  var app = new EmberApp(defaults, {
    // Add options here
    fingerprint: {
      exclude: ['ckeditor'],
      ignore: ['ckeditor']
    }
  });
```

**Comment by ChrisHonniball** [comment]
> Found it to be fingerprinting... Updated `ember-cli-build.js` to include:
> 
> ``` javascript
>   var app = new EmberApp(defaults, {
>     // Add options here
>     fingerprint: {
>       exclude: ['ckeditor'],
>       ignore: ['ckeditor']
>     }
>   });
> ```

Using these settings allows for you to revert back to the included:

``` javascript
  treeForPublic: function(tree) {
    return new Funnel(this.project.bowerDirectory + '/ckeditor', {
      srcDir: '/',
      destDir: '/assets/ckeditor'
    });
  },
```

and build times are back to normal. The updates are in my fork with some additional alterations that I've made for my specific situation. You can check them out [here](https://github.com/ChrisHonniball/ember-ckeditor) and if you want them I can submit a pull request.

**Comment by ghost** [comment]
+1 on this issue...Still having it too after trying the fixes... What can I do to solve this problem correctly? I need fast build times for production..

**Comment by Leooo** [comment]
+100 on this issue, took ages to see why my heroku build was failing

**Comment by ChrisHonniball** [comment]
Using:

``` javascript
  var app = new EmberApp(defaults, {
    // Add options here
    fingerprint: {
      exclude: ['ckeditor'],
      ignore: ['ckeditor']
    }
  });
```

I'm currently running my fork of this repo and build times are normal.

**Comment by Leooo** [comment]
Yes @ChrisHonniball this solves the issue thanks.. if we think of looking at this thread.

**Comment by ChrisHonniball** [comment]
👍  I've added a comment about it to [my repo](https://github.com/ChrisHonniball/ember-ckeditor) however I've branched off quite a bit from this repo so I'm not sure if the updates will be needed/wanted in this repo.
